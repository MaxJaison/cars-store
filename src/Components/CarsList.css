.cars {
  width: 100%;
  margin-left: 24px;
}

.card {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  padding: 24px;
  border: 1px solid #ededed;
}

.image {
  width: 120px;
  height: 100px;
}

.content {
  display: flex;
}

.grey-square {
  width: 120px;
  height: 100px;
  margin-right: 20px;
  background-color: #ededed;
}

.grey-line {
  width: 200px;
  margin-bottom: 10px;
  height: 30px;
  background-color: #ededed;
}

.grey-line:not(:first-child) {
  height: 20px;
  width: 150px;
}
