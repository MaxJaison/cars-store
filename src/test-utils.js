import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
