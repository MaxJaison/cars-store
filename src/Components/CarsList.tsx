import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import "./CarsList.css";
import { useDispatch, useSelector } from "react-redux";
import { Car, RootState } from "../types";
import { fetchCars } from "../actionCreators/carsActions";
import { Card, CardContent, CardMedia } from "@material-ui/core";
import Pagination from "@material-ui/lab/Pagination";
import EmptyCard from "./EmptyCard";

const CarsList = () => {
  const dispatch = useDispatch();
  const data = useSelector((state: RootState) => state.cars);
  const color = useSelector((state: RootState) => state.color);
  const manufacture = useSelector((state: RootState) => state.manufacture);
  const sortBy = useSelector((state: RootState) => state.sortBy);
  const page = useSelector((state: RootState) => state.page);

  useEffect(() => {
    dispatch(fetchCars(1, []));
  }, [dispatch]);

  const handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
    let filters = [];
    if (color) {
      filters.push({ name: "color", value: color });
    }

    if (manufacture) {
      filters.push({ name: "manufacturer", value: manufacture });
    }

    if (sortBy !== "none") {
      filters.push({ name: "sort", value: sortBy });
    }

    dispatch(fetchCars(value, filters));
  };

  return (
    <div className="cars">
      {data.pending ? (
        <EmptyCard />
      ) : (
        <>
          <h2>Available Cars</h2>
          <p>
            Showing{" "}
            {data.carsData.totalCarsCount < 10
              ? data.carsData.totalCarsCount
              : data.carsData.cars.length === 10
              ? 10
              : data.carsData.totalCarsCount % 10}{" "}
            of {data.carsData.totalCarsCount} results
          </p>
          {data.carsData.cars.map((car: Car) => {
            return (
              <Card className="card" key={car.stockNumber}>
                <CardMedia
                  className="image"
                  image={car.pictureUrl}
                  title={car.modelName}
                />
                <CardContent>
                  <h2>
                    {car.manufacturerName} {car.modelName}
                  </h2>
                  <p>
                    Stock # {car.stockNumber} - {car.mileage.number / 1000}{" "}
                    {car.mileage.unit} - {car.fuelType} - {car.color}
                  </p>
                  <Link className="link" to={`/cars/${car.stockNumber}`}>
                    View Details
                  </Link>
                </CardContent>
              </Card>
            );
          })}
          <div>
            <Pagination
              count={data.carsData.totalPageCount}
              page={page}
              onChange={handleChange}
              defaultPage={1}
            />
          </div>
        </>
      )}
    </div>
  );
};

export default CarsList;
