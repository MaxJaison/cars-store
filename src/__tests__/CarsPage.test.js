import { render } from "../test-utils";
import React from "react";
import CarsList from "../Components/CarsList";

test("renders CarsPage", () => {
  const { container } = render(<CarsList />, {});
  expect(container).toMatchSnapshot();
});
