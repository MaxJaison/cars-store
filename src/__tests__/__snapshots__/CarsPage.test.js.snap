// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CarsPage 1`] = `
<div>
  <div
    class="cars"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root content"
      >
        <div
          class="grey-square"
        />
        <div>
          <div
            class="grey-line"
          />
          <div
            class="grey-line"
          />
          <div
            class="grey-line"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
